packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -99,124 +107,330 @@ const defaultProps = {
   value: undefined,
 };
 
-// eslint-disable-next-line react/prefer-stateless-function
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
+
+const dateState = { day: '', month: '', year: '' };
+
+function DatePickerInput(props) {
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayFocus, setDayFocus] = useState(null);
+  const [monthFocus, setMonthFocus] = useState(null);
+  const [yearFocus, setYearFocus] = useState(null);
+
+  const setDayRef = useCallback(node => {
+    setDayFocus(node);
+  }, []);
+
+  const setMonthRef = useCallback(node => {
+    setMonthFocus(node);
+  }, []);
+
+  const setYearRef = useCallback(node => {
+    setYearFocus(node);
+  }, []);
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
-  }
+  const {
+    buttonRefCallback,
+    inputAttributes,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onChange,
+    onClick,
+    onFocus,
+    onButtonFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    selectedDate,
+    value,
+    ...customProps
+  } = props;
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const localeFormat = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+  const dateValue = DateUtil.convertToISO8601(value, DateUtil.getFormatByLocale(intl.locale));
+  const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
+  const dateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const placeholderValues = DateUtil.getPlaceholderValues(localeFormat, placeholder);
+
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
+    }
+
+    // Clears the inputs when appropriate (e.g. excluded, out of range)
+    if (!selectedDate) {
+      dateDispatch({ day: '', month: '', year: '' });
     }
-  }
+  }, [selectedDate, shouldShowPicker, onClick]);
 
-  handleOnButtonClick(event) {
-    const attributes = { ...this.props.inputAttributes };
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  useEffect(() => {
+    if (DateUtil.isValidDate(value, dateFormat)) {
+      const parseDate = {
+        dayValue: dateValue.substring(8, 10),
+        monthValue: dateValue.substring(5, 7),
+        yearValue: dateValue.substring(0, 4),
+      };
```

Comment:
This feels like it should be using the same parsing logic that was used in `getPlaceholderValues`